src/main/java/model/Wallet.java

```diff
@@ -0,0 +1,60 @@
+package model;
+
+import exception.InvalidMoneyAmountException;
+
+public class Wallet {
+    private int money;
+
+    /**
+     * @param money: integer
+     * @pre money >= 0
+     */
+    public Wallet(int money) {
+        this.money = Math.max(money, 0);
+    }
+
+    public Wallet() {
+        this.money = 100;
+    }
+
+    /**
+     * @param amount: int
+     * @pre amount >= 0
+     */
+    public void put(int amount) throws InvalidMoneyAmountException {
```

Comment:
Ich würde persönlich nur eine Funktion "add" oder so machen, statt put() und withdraw(). Im Endeffekt machen beide "this.money += amount", warum also zwei Funktionen dafuer definieren? Das erhoeht ja nur den Aufwand fuer den, der die Funktion aufruft, was oefters passiert als dass diese Funktion programmiert wird